pkg/api/v1/types.go

```diff
@@ -1107,3 +1110,12 @@ const (
 	// Whether all components were ready
 	KubeVirtConditionReady KubeVirtConditionType = "Ready"
 )
+
+// ---
+// +k8s:openapi-gen=true
+type EvictionPolicy string
+
+const (
+	EvictionPolicyPowerOff    EvictionPolicy = "PowerOff"
+	EvictionPolicyLiveMigrate EvictionPolicy = "LiveMigrate"
```

Comment:
Yeah. I learned yesterday that ther  is a k8s  PR which will enable webhooks for k8s. However that will probably take some time until we have it broadly available. Fabian and I therefore thought about redesigning the whole section into more something like this:

```yaml
evictionPolicy:
  taints:
    - key: kubernetes.io/noSchedule
      effect: NoSchedule
      tolerationSeconds: 123s
      strategy: LiveMigrate
```

This would then better reflect what we really do. But it might still not properly express that it might not happen. Do you have an idea there?

Reply:
Another maybe nicer option would be to extend the API with `spec.tolerations` in general and add an optional strategy there:

```yaml
spec:
  tolerations:
  - key: kubernetes.io/noSchedule
    effect: NoSchedule
    tolerationSeconds: 123s
    strategy: LiveMigrate # optional, defaults to PowerOff
```